src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -190,4 +198,44 @@ private function createChildDonation( Donation $donation, PayPalNotificationRequ
 		}
 		return $childDonation;
 	}
+
+	private function newDonorFromRequest( PayPalNotificationRequest $request ): Donor {
+		return new Donor(
+			$this->newPersonNameFromRequest( $request ),
+			$this->newPhysicalAddressFromRequest( $request ),
+			$request->getPayerEmail()
+		);
+	}
+
+	private function newPersonNameFromRequest( PayPalNotificationRequest $request ): PersonName {
+		$name = PersonName::newPrivatePersonName();
+		$name->setFirstName( $request->getPayerFirstName() );
+		$name->setLastName( $request->getPayerLastName() );
+		$name->freeze();
```

Comment:
The `assertNoNullFields` is part of normal `FreezableValueObject` initialization. As those do not have a real constructor, and instead require you to call setters, it's easy to forget to call a setter. And one can add a new field, and then not have the existing code that is wrongly not calling the new setter still work, though incorrectly.

If there is a field that can be null for valid reasons, then this does not work. Should probably do something else in that case, but I figure we have more pressing matters to get to.